File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/BookManagement.cpp
#include <string>
#include <map>

class BookManagement {
private:
    std::map<std::string, int> inventory;

public:
    BookManagement() {}

    void add_book(const std::string& title, int quantity = 1) {
        inventory[title] += quantity;
    }

    void remove_book(const std::string& title, int quantity) {
        if (inventory.find(title) == inventory.end() || inventory[title] < quantity) {
            throw false;
        }
        inventory[title] -= quantity;
        if (inventory[title] == 0) {
            inventory.erase(title);
        }
    }

    std::map<std::string, int> view_inventory() {
        return inventory;
    }

    int view_book_quantity(const std::string& title) {
        auto it = inventory.find(title);
        if (it == inventory.end()) {
            return 0;
        }
        return it->second;
    }
};

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_py/code_output/BookManagement.py
class BookManagement:
    def __init__(self):
        self.inventory = {}

    def add_book(self, title, quantity):
        # Equivalent to Java's containsKey/get/put logic, using dict.get for idiomatic Python
        self.inventory[title] = self.inventory.get(title, 0) + quantity

    def remove_book(self, title, quantity):
        if title not in self.inventory or self.inventory[title] < quantity:
            raise Exception("Invalid operation")
        
        new_quantity = self.inventory[title] - quantity
        if new_quantity == 0:
            del self.inventory[title]
        else:
            self.inventory[title] = new_quantity

    def view_inventory(self):
        # Returns a shallow copy of the dictionary, identical to new HashMap<>(inventory)
        return self.inventory.copy()

    def view_book_quantity(self, title):
        # Equivalent to Java's getOrDefault
        return self.inventory.get(title, 0)